wgengine/netstack/netstack.go

```diff
@@ -161,19 +162,17 @@ func (ns *Impl) updateIPs(nm *netmap.NetworkMap) {
 }
 
 func (ns *Impl) dialTCP(address string) (*gonet.TCPConn, error) {
-	remoteIPStr, remotePortStr, err := net.SplitHostPort(address)
+	remoteIPPort, err := netaddr.ParseIPPort(address)
 	if err != nil {
 		return nil, fmt.Errorf("could not parse IP:port: %w", err)
 	}
-	remotePort, _ := strconv.Atoi(remotePortStr)
-	remoteIP := net.ParseIP(remoteIPStr)
 	remoteAddress := tcpip.FullAddress{
 		NIC:  nicID,
-		Addr: tcpip.Address(remoteIP),
-		Port: uint16(remotePort),
+		Addr: tcpip.Address(remoteIPPort.TCPAddr().IP),
```

Comment:
`remoteIPPort.IP.IPAddr().IP`

(what a mouthful!)

Reply:
Is there a difference between this and the one I wrote? If so, what is it?

Reply:
Just a minor efficiency/intent difference. Little weird to call TCPAddr just to throw away the port. All you want is the IP part, so IPAddr() is what you want to just convert to std's IP type.